### ISSUE TITLE
Deno raising `TypeError` instead of `Deno.errors.ConnectionRefused`

### ISSUE THREAD

**Issue by jflatow** [denoland/deno#20394]
I first encountered something like this with the embedded runtime and reported it in #16737. So I vaguely know why it happens, but now its happening on network errors with the official `deno` cli. Can I really be the only one hitting this? Here's a minimal example showing what I'm talking about:

```
> await fetch('http://localhost:90210').catch(e => e instanceof Deno.errors.ConnectionRefused)
false
```

Here's my `deno --version`: 
```
deno 1.31.3 (release, x86_64-apple-darwin)
v8 11.0.226.19
typescript 4.9.4
```

**Comment by piscisaureus** [comment]
I think this is (somewhat unfortunately) required by the [fetch() spec](https://fetch.spec.whatwg.org/#fetch-method) (see § 5.6 step 12.3).

**Comment by jflatow** [comment]
> I think this is (somewhat unfortunately) required by the [fetch() spec](https://fetch.spec.whatwg.org/#fetch-method) (see § 5.6 step 12.3).

Ah I guess that's fair then, good catch, thanks. Should it somehow be documented better? Perhaps this issue is enough for posterity.

It might be nice for `Deno.errors` to provide a bunch of predicates if this situation becomes more pervasive (I've already encountered a few times as I referenced earlier, and end up writing these in my application code). For instance:

```
function isRefused(e: Error): boolean {
  return /refused/i.test(e.message);
}
```

But probably more robust...
